api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,313 @@
+	var wfrId string
+
+	err = w.RegisterWorkflow(
+		&worker.WorkflowJob{
+			On:          worker.Events(m.eventName),
+			Name:        workflowName,
+			Description: "This is part of the monitoring system for testing the readiness of this Hatchet installation.",
+			Steps: []*worker.WorkflowStep{
+				worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+
+					input := &probeEvent{}
+
+					err = ctx.WorkflowInput(input)
+					if err != nil {
+						return nil, err
+					}
+
+					wfrId = ctx.WorkflowRunId()
+
+					if input.UniqueStreamId == "" {
+						return nil, fmt.Errorf("uniqueStreamId is required")
+					}
+
+					if input.UniqueStreamId != streamValue {
+						return nil, fmt.Errorf("uniqueStreamId does not match stream value")
+					}
+
+					ctx.StreamEvent([]byte("This is a stream event for step-one"))
+
+					stepChan <- "step-one"
+
+					return &stepOneOutput{
+						Message:        "This is a message for step-one",
+						UniqueStreamId: streamValue,
+					}, nil
+				},
+				).SetName("step-one"),
+				worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+					input := &probeEvent{}
+					err = ctx.StepOutput("step-one", input)
+
+					if err != nil {
+						return nil, err
+					}
+
+					if input.UniqueStreamId == "" {
+						return nil, fmt.Errorf("uniqueStreamId is required")
+					}
+
+					if input.UniqueStreamId != streamValue {
+						return nil, fmt.Errorf("uniqueStreamId does not match stream value")
+					}
+
+					ctx.StreamEvent([]byte("This is a stream event for step-two"))
+					stepChan <- "step-two"
+
+					return &stepOneOutput{
+						Message:        "This is a message for step-two",
+						UniqueStreamId: streamValue,
+					}, nil
+				}).SetName("step-two").AddParents("step-one"),
+			},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error registering workflow: %w", err)
+	}
+
+	go func() {
+		err = c.Subscribe().StreamByAdditionalMetadata(ctx, streamKey, streamValue, func(event client.StreamEvent) error {
+			events <- string(event.Message)
+
+			return nil
+		})
+		if err != nil {
+			errors <- fmt.Errorf("error subscribing to stream: %w", err)
+		}
+	}()
+
+	go func() {
+		testEvent := probeEvent{
+			UniqueStreamId: streamValue,
+		}
+
+		err := c.Event().Push(
+			ctx,
+			m.eventName,
+			testEvent,
+			client.WithEventMetadata(map[string]string{
+				streamKey: streamValue,
+			},
+			),
+		)
+
+		if err != nil {
+			errors <- fmt.Errorf("error pushing event: %w", err)
+		}
+	}()
+
+	cleanupWorker, err := w.Start()
+
+	if err != nil {
+		return nil, fmt.Errorf("error starting worker: %w", err)
+	}
+
+	cleanup := func() {
+		err := cleanupWorker()
+		if err != nil {
+			m.l.Error().Msgf("error cleaning up worker: %s", err)
+		}
+		defer func() {
+			if wfrId == "" {
+				m.l.Warn().Msg("workflow run id was never set for probe")
+				return
+			}
+
+			ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
This timeout seems quite low for a PG delete -- perhaps something like 10 seconds?